### PR TITLE
tests/runk: fix the "run ps command" flaky test

### LINKED ISSUE #8975: CI: run-runk failing intermittently on the `test ps command` checker
I've noticed intermittent failures on run-runk job, often you run a 2nd time and the test pass.

It is in the `test ps command` that sometimes return 2 instead of the expected 3 lines:
```
[runk-tests.sh:78] INFO: ctr pull'd: docker.io/library/busybox:latest
start container with runk
exec process in a container
test ps command
ps output:
PID     INFO
3795    -
ps output lines: 2
[runk-tests.sh:48] ERROR: unexpected ps command output
```

Examples of fails in https://github.com/kata-containers/kata-containers/actions/runs/7726958090/job/21065554095?pr=8953 and https://github.com/kata-containers/kata-containers/actions/runs/7714184548/job/21028062099?pr=8964

### PR DESCRIPTION
**Pull request by wainersm**
The "run ps command" test of the runk suite has failed intermittently. So here I'm fixing that but before I converted it to bats entirely; it should help on future maintenance of the test suite besides of improving the reporting text. 

Fixes #8975

### CODE REVIEW

**Comment by wainersm** [comment]
hi @ManaSugi , mind to review it? Also on commit https://github.com/kata-containers/kata-containers/pull/9009/commits/69ea7f11ddb3be2f7d02f41f8bf5f5cb44afdb73 I added you as a co-author, so could you please check the signed-off string is accurate?

File: tests/integration/runk/runk-tests.bats [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+	# even if the container already is stopped. Hence, a low-level runtime
+	# should allow kill --all regardless of the container state like runc.
+	echo "test kill --all is allowed regardless of the container state"
+	# Check kill should fail because the container is paused
+	stop_container
+	run sudo ctr t kill --signal SIGKILL ${CONTAINER_ID}
+	[ $status -eq 1 ]
+	# Check  kill --all should not fail
+	sudo ctr t kill --signal SIGKILL --all "${CONTAINER_ID}"
+}
```
Comment (ManaSugi) [inline tests/integration/runk/runk-tests.bats:123]: an empty line

**Review by ManaSugi — COMMENTED** [review]
@wainersm Sorry for the late review. my sign-off is good, thanks!

File: tests/integration/runk/runk-tests.bats [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+# SPDX-License-Identifier: Apache-2.0
+#
+# This test will validate runk with containerd
+
+load "${BATS_TEST_DIRNAME}/../../common.bash"
+load "${BATS_TEST_DIRNAME}/../../metrics/lib/common.bash"
+
+setup_file() {
+	export RUNK_BIN_PATH="/usr/local/bin/runk"
+	export TEST_IMAGE="docker.io/library/busybox:latest"
```
Comment (danmihai1) [inline tests/integration/runk/runk-tests.bats:14]: Is quay.io/prometheus/busybox preferrable, to work around image repository limits?

**Review by danmihai1 — APPROVED** [review]

File: tests/integration/runk/runk-tests.bats [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+		${TEST_IMAGE} \
+		${CONTAINER_ID}
+	read CID PID STATUS <<< $(sudo ctr t ls | grep ${CONTAINER_ID})
+	# Check the pid is consistent
+	[ "${PID}" == "$(cat "${PID_FILE}")" ]
+	# Check the container status is RUNNING
+	[ "${STATUS}" == "RUNNING" ]
+}
+
+teardown() {
```
Comment (GabyCT) [inline tests/integration/runk/runk-tests.bats:37]: mmm why not to use https://github.com/kata-containers/kata-containers/blob/main/tests/common.bash#L182?

File: tests/integration/runk/runk-tests.bats (hunk 1 above)
Comment (wainersm) [reply tests/integration/runk/runk-tests.bats:123]: Fixed.

File: tests/integration/runk/runk-tests.bats (hunk 3 above)
Comment (wainersm) [reply tests/integration/runk/runk-tests.bats:37]: hi @GabyCT , you mean use `check_processes` to ensure that shimv2 process really got killed?

**Comment by wainersm** [comment]
Updates:
 * rebased code to main
 * replaced busybox image as suggested by @danmihai1 
 * added extra empty line as suggested by @ManaSugi

**Review by ManaSugi — APPROVED** [review]

**Comment by ManaSugi** [comment]
lgtm, thanks @wainersm !

**Pull request merged**